api/fees/fees.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/chain"
+)
+
+type Fees struct {
+	data           *FeesData
+	bft            bft.Committer
+	backtraceLimit uint32 // The max number of blocks to backtrace.
+}
+
+func New(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedCacheSize uint32) *Fees {
```

Comment:
Would you consider make `fixedCacheSize` a internal constant?

Reply:
That would make the tests break because I would not be able to control the cache size to test this (right now we use 10 blocks, so if having 1024 every I would have issues to validate scenarios)

Reply:
I might create a constant at `api.go`, that would work in any case, please confirm that is ok for you

Reply:
Added to `api.go`